salt/states/saltmod.py

```diff
@@ -689,6 +732,137 @@ def runner(name, **kwargs):
+                                           __orchestration_jid__=jid,
+                                           __env__=__env__,
+                                           full_return=True,
+                                           **(runner_config['kwarg']))
+
+    outputs = _parallel_map(call_runner, runners)
+
+    success = six.moves.reduce(
+        lambda x, y: x and y,
+        [not ('success' in out and not out['success']) for out in outputs],
+        True)
+
+    def find_new_and_old_in_changes(data, prefix):
+        if isinstance(data, dict) and data:
+            if 'new' in data and 'old' in data:
+                return [(prefix, {'new': data['new'], 'old': data['old']})]
+            else:
+                return [
+                    change_item
+                    for key, value in six.iteritems(data)
+                    for change_item in find_new_and_old_in_changes(
+                        value, prefix + '[' + str(key) + ']')
+                ]
+        if isinstance(data, list) and list:
+            return [
+                change_item
+                for index, value in six.moves.zip(
+                    six.moves.range(len(data)), data)
+                for change_item in find_new_and_old_in_changes(
+                    value, prefix + '[' + str(index) + ']')
+            ]
+        else:
+            return []
+    def find_changes(data, prefix):
+        if isinstance(data, dict) and data:
+            if 'changes' in data:
+                return find_new_and_old_in_changes(data['changes'], prefix)
+            else:
+                return [
+                    change_item
+                    for key, value in six.iteritems(data)
+                    for change_item in find_changes(
+                        value, prefix + '[' + str(key) + ']')
+                ]
+        else:
+            return []
+    def find_changes_in_output(output, index):
+        try:
+            data = output['return']['data']
+        except KeyError:
+            data = {}
+        return find_changes(data, '[' + str(index) + ']')
+    changes = dict([
+        change_item
+        for change_items in six.moves.map(
+            find_changes_in_output, outputs, six.moves.range(len(outputs)))
+        for change_item in change_items
+    ])
+
+    def generate_comment(index, out):
+        runner_failed = 'success' in out and not out['success']
+        runner_return = out.get('return')
+        comment = (
+            'Runner ' + str(index) + ' was '
+            + ('not ' if runner_failed else '')
+            + 'successful and returned '
+            + (str(runner_return) if runner_return else ' nothing') + '.')
+        return comment
+    comment = '\n'.join(six.moves.map(generate_comment,
+                                      six.moves.range(len(outputs)),
+                                      outputs))
+
+    ret = {
+        'name': name,
+        'result': success,
```

Comment:
The `result` field in the highstate output and the `success` field in the job output are separate things and should be handled separately. `success` indicates whether there were any _internal_ Salt failures, whereas the `result` field indicates whether any individual state functions failed.